sdk/eventhub/Azure.Messaging.EventHubs/src/BatchOptions.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs
+{
+    /// <summary>
+    ///   The set of options that can be specified to influence the way in which an event batch
+    ///   behaves and is sent to the Event Hubs service.
+    /// </summary>
+    ///
+    public class BatchOptions : SendOptions
+    {
+        /// <summary>The requested maximum size to allow for the batch, in bytes.</summary>
+        private long? _maximumSizeInBytes = null;
+
+        /// <summary>
+        ///   The maximum size to allow for a single batch of events, in bytes.
+        /// </summary>
+        ///
+        /// <value>
+        ///   The desired limit, in bytes, for the size of the associated event batch.  If <c>null</c>,
+        ///   the maximum size allowed by the active transport will be used.
+        /// </value>
+        ///
+        public long? MaximumizeInBytes
+        {
+            get => _maximumSizeInBytes;
+
+            set
+            {
+                if (value.HasValue)
+                {
+                    Guard.ArgumentAtLeast(nameof(MaximumizeInBytes), value.Value, EventHubProducer.MinimumBatchSizeLimit);
+                }
+
+                _maximumSizeInBytes = value;
+            }
+        }
+
+        /// <summary>
+        ///   Creates a new copy of the current <see cref="BatchOptions" />, cloning its attributes into a new instance.
+        /// </summary>
+        ///
+        /// <returns>A new copy of <see cref="BatchOptions" />.</returns>
+        ///
+        internal BatchOptions Clone() =>
+            new BatchOptions
+            {
+                PartitionKey = this.PartitionKey,
+                _maximumSizeInBytes = this.MaximumizeInBytes
+            };
+
+        /// <summary>
+        ///   Determines whether the specified <see cref="System.Object" />, is equal to this instance.
```

Comment:
nit: I'm not very knowledgeable about the English language, so I might be wrong, but I think we should drop this comma.

Reply:
You're correct.  No idea how that got there.

Reply:
13 places through the project.  Yikes!